templates/advanced/api/services/Ember.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * Ember service
+ *
+ * @module Ember
+ */
+
+var _ = require( 'lodash' ),
+	pluralize = require( 'pluralize' );
+
+var Ember = {
+	linkAssociations: function ( model, records ) {
+		if ( !Array.isArray( records ) ) records = [ records ];
+		var modelPlural = pluralize( Ember.emberizeModelIdentity( model ) );
+
+		return _.map( records, function ( record ) {
+			var links = {};
+			_.each( model.associations, function ( assoc ) {
+				if ( assoc.type === "collection" ) {
+					links[ assoc.alias ] = sails.config.blueprints.prefix + "/" + modelPlural + "/" + record.id + "/" + assoc.alias;
+				}
+			} );
+			if ( _.size( links ) > 0 ) {
+				record.links = links;
+			}
+			return record;
+		} );
+	},
+
+	emberizeModelIdentity: function ( model ) {
+		return model.globalId.replace( /(?:^\w|[A-Z]|\b\w|\s+)/g,
+			function ( match, index ) {
+				if ( +match === 0 ) return ""; // or if (/\s+/.test(match)) for white spaces
+				return index === 0 ? match.toLowerCase() : match.toUpperCase();
+			} );
+	},
+
+	/**
+	 * Prepare records and populated associations to be consumed by Ember's DS.RESTAdapter
+	 *
+	 * @param {Collection} model Waterline collection object (returned from parseModel)
+	 * @param {Array|Object} records A record or an array of records returned from a Waterline query
+	 * @param {Associations} associations Definition of the associations, from `req.option.associations`
+	 * @param {Boolean} sideload Sideload embedded records or reduce them to primary keys?
+	 * @return {Object} The returned structure can be consumed by DS.RESTAdapter when passed to res.json()
+	 */
+	buildResponse: function ( model, records, associations, sideload, associatedRecords ) {
+		sideload = sideload || false;
+		var plural = Array.isArray( records ) ? true : false;
+
+		var emberModelIdentity = Ember.emberizeModelIdentity( model );
+		var modelPlural = pluralize( emberModelIdentity );
+		var documentIdentifier = plural ? modelPlural : emberModelIdentity;
+		var json = {};
+
+		json[ documentIdentifier ] = plural ? [] : {};
```

Comment:
Not sure if this will simplify your code at all but the Ember Data RESTSerializer allows you can return a single resource wrapped in an array. Returning an object without the array still works but it is considered more of a legacy thing https://github.com/emberjs/data/blob/master/packages/ember-data/lib/serializers/rest_serializer.js#L283-L287